ni_measurementlink_service/measurement/info.py

```diff
@@ -67,26 +67,107 @@ class TypeSpecialization(enum.Enum):
+UInt64TypeInfo = DataTypeInfo(type_pb2.Field.TYPE_UINT64, False)
+FloatTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_FLOAT, False)
+DoubleTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_DOUBLE, False)
+BooleanTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_BOOL, False)
+StringTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_STRING, False)
+PinTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_STRING, False, TypeSpecialization.Pin)
+PathTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_STRING, False, TypeSpecialization.Path)
+EnumTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_ENUM, False, TypeSpecialization.Enum)
+
+Int32Array1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_INT32, True)
+Int64Array1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_INT64, True)
+UInt32Array1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_UINT32, True)
+UInt64Array1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_UINT64, True)
+FloatArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_FLOAT, True)
+DoubleArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_DOUBLE, True)
+BooleanArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_BOOL, True)
+StringArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_STRING, True)
+PinArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_STRING, True, TypeSpecialization.Pin)
+PathArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_STRING, True, TypeSpecialization.Path)
+EnumArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_ENUM, True, TypeSpecialization.Enum)
+
+
+class DataTypeInfoLookup:
+    """Lookup information about a DataType."""
+
+    _DATATYPE_TO_DATATYPEINFO_LOOKUP = {
+        DataType.Int32: Int32TypeInfo,
+        DataType.Int64: Int64TypeInfo,
+        DataType.UInt32: UInt32TypeInfo,
+        DataType.UInt64: UInt64TypeInfo,
+        DataType.Float: FloatTypeInfo,
+        DataType.Double: DoubleTypeInfo,
+        DataType.Boolean: BooleanTypeInfo,
+        DataType.String: StringTypeInfo,
+        DataType.Pin: PinTypeInfo,
+        DataType.Path: PathTypeInfo,
+        DataType.Enum: EnumTypeInfo,
+        DataType.Int32Array1D: Int32Array1DTypeInfo,
+        DataType.Int64Array1D: Int64Array1DTypeInfo,
+        DataType.UInt32Array1D: UInt32Array1DTypeInfo,
+        DataType.UInt64Array1D: UInt64Array1DTypeInfo,
+        DataType.FloatArray1D: FloatArray1DTypeInfo,
+        DataType.DoubleArray1D: DoubleArray1DTypeInfo,
+        DataType.BooleanArray1D: BooleanArray1DTypeInfo,
+        DataType.StringArray1D: StringArray1DTypeInfo,
+        DataType.PinArray1D: PinArray1DTypeInfo,
+        DataType.PathArray1D: PathArray1DTypeInfo,
+        DataType.EnumArray1D: EnumArray1DTypeInfo,
+    }
+
+    @staticmethod
+    def get_type_info(data_type: DataType) -> DataTypeInfo:
+        """Get information about a DataType."""
+        if data_type not in DataTypeInfoLookup._DATATYPE_TO_DATATYPEINFO_LOOKUP:
+            raise Exception(f"Data type information not found '{data_type}'")
+        data_type_info = DataTypeInfoLookup._DATATYPE_TO_DATATYPEINFO_LOOKUP[data_type]
```

Comment:
This does two dictionary lookups when it only needs one.

```suggestion
        data_type_info = DataTypeInfoLookup._DATATYPE_TO_DATATYPEINFO_LOOKUP.get(data_type)
        if data_type_info is None:
            raise Exception(f"Data type information not found '{data_type}'")
```

I'm not really worried about performance because this lookup is only done at initialization time, but IMO using get() is a "best practice" and is also simpler than doing two lookups.

Reply:
It still does two lookups. Delete the 2nd lookup.

Reply:
Oops. Fixed.